### DISCUSSION TITLE
[A/L] (2021/22) Relazioni - Esercizio 3

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#275] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152318278-9b8a18a9-8445-4eaf-a716-90dec81b1ba0.png)

**Comment by GattoBrizzolato** [comment]
1. V
2. F

**Comment by LuginiAndrea** [comment]
Il primo caso possiamo riscriverlo come 
$$R^* = R \cup S \implies aR^*b \Longleftrightarrow aRb \vee aSb$$
Riflessività: rispettata in quanto $aRa$ implica $aR^*a$
Simmetria: rispettata in quanto $aRb \vee aSb \Longrightarrow bRa \vee bSa \Longrightarrow bR^*a$
Transitività: Ipotizziamo 
$$\exists aRb \wedge \exists bSc \wedge \nexists aRc \wedge \nexists aSc \implies \exists aR^*b \wedge \exists bR^*c \wedge \nexists aR^*c$$
Quindi la relazione non è transitiva, quindi non si tratta di equivalenza

Analizziamo il secondo caso:
$$R^* = aR^*b \Longleftrightarrow aRb \wedge \nexists aSb$$

Riflessività: In quanto relazioni di equivalenza sullo stesso insieme, $$\forall a \in A \exists aRa \wedge\exists aSa \implies \exists \left(a, a\right) \in R \wedge \exists \left(a,a\right) \in S \implies \nexists \left(a,a\right) \in R^* \implies \nexists aR^*a$$ 
Quindi la riflessività non è rispettata e quindi non si tratta di una relazione d'equivalenza

**Comment by Exyss** [comment]
## $R \cup S$
Per comodità, ridenominiamo $R \cup S$ come $\sim$ ed affermiamo che $x \sim y\iff (x,y) \in R \lor (x,y) \in S$

### Riflessività :heavy_check_mark: 
$a \in A \implies (a,b) \in R \land (a,a) \in S \implies a \sim b$

### Simmetria :heavy_check_mark: 
$a \sim b \implies (a,b) \in R \lor (a,b) \in S \implies (b,a) \in R \lor (b,a) \in S \implies b \sim a$

### Transitività :no_entry_sign: 
$a \sim b \land b \sim c \implies ((a,b) \in R \lor (a,b) \in S) \land ((b,c) \in R \lor (b,c) \in S)

A questo punto, consideriamo il caso in cui
$$(a,b) \in R \land (b,c) \notin R \land (a,b) \in S \land (b.c) \notin S \implies (a,c) \notin R \land (a,c) \notin S \implies a \not\sim c$$


--------

## $R - S$

Per comodità, ridenominiamo $R - S$ come $\sim$ ed affermiamo che $x \sim y \iff (x,y) \in R \land (x,y) \notin S$

### Riflessività :no_entry_sign: 
$a \in A \implies (a,a) \in R \land (a,a) \in S \implies a \not\sim a$

_(simmetria e transitività omesse)_
